packages/amplify-category-auth/resources/lambda-function.js

```diff
@@ -6,12 +6,26 @@ exports.handler = async function (event, context) {
     const userPoolId = event.ResourceProperties.userpoolId;
     const lambdaConfig = event.ResourceProperties.lambdaConfig;
     const config = {};
+    const cognitoClient = new aws.CognitoIdentityServiceProvider();
+    const userPoolConfig = await cognitoClient.describeUserPool({ UserPoolId: userPoolId }).promise();
+    // convert describe userPool return object to updateUserpool input
+    const userPoolParams = userPoolConfig.UserPool;
+    delete userPoolParams.Id;
+    delete userPoolParams.Name;
+    delete userPoolParams.LastModifiedDate;
+    delete userPoolParams.CreationDate;
+    delete userPoolParams.SchemaAttributes;
+    delete userPoolParams.EstimatedNumberOfUsers;
+    delete userPoolParams.UsernameConfiguration;
+    delete userPoolParams.Arn;
+    delete userPoolParams.AdminCreateUserConfig.UnusedAccountValidityDays;
     lambdaConfig.forEach(lambda => (config[`${lambda.triggerType}`] = lambda.lambdaFunctionArn));
     if (event.RequestType == 'Delete') {
-      const authParams = { UserPoolId: userPoolId, LambdaConfig: {} };
-      const cognitoclient = new aws.CognitoIdentityServiceProvider();
       try {
-        const result = await cognitoclient.updateUserPool(authParams).promise();
+        const authParams = userPoolParams;
```

Comment:
Why the assignment? it is byref, pointing to the same object, if we need a copy of it, then clone it, otherwise use `userPoolParams`